File: java/src/test/java/codingdojo/LongRangeForecastsTest.java
package codingdojo;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Map;
import java.util.function.Function;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class LongRangeForecastsTest {

    @Test
    void longRangeForecast() {
        var quarter = "Q3";
        Function<Long, Boolean> scorerLookup = daysForward -> {
            return Boolean.TRUE;
        };
        var actual = LongRangeForecasts.getIceCreamForecast(quarter, scorerLookup);
        StringBuilder printedResult = printIceCreamForecast(actual);

        Approvals.verify(printedResult);
    }

    private static StringBuilder printIceCreamForecast(Map<IceCream, Integer> actual) {
        var printedResult = new StringBuilder();
        printedResult.append("{");
        var sortedKeys = new ArrayList<IceCream>(actual.keySet());
        Collections.sort(sortedKeys);
        for (IceCream key : sortedKeys) {
            printedResult.append(key);
            printedResult.append(":");
            printedResult.append(actual.get(key));
            printedResult.append(", ");
        }
        printedResult.append("}");
        return printedResult;
    }
}

File: kotlin/src/test/kotlin/org/codingdojo/LongRangeForecastsTest.kt
package org.codingdojo

import org.approvaltests.Approvals
import org.junit.jupiter.api.Test
import java.util.*

class LongRangeForecastsTest {

    @Test
    fun longRangeForecast() {
        val quarter = "Q3"
        val actual = LongRangeForecasts().longRangeForecast(quarter)
        val printedResult = printIceCreamForecast(actual)
        Approvals.verify(printedResult)
    }

    private fun printIceCreamForecast(actual: Map<IceCream, Int>): StringBuilder {
        val printedResult = StringBuilder()
        printedResult.append("{")
        val sortedKeys = ArrayList(actual.keys)
        Collections.sort(sortedKeys)
        for (key in sortedKeys) {
            printedResult.append(key)
            printedResult.append(":")
            printedResult.append(actual[key])
            printedResult.append(", ")
        }
        printedResult.append("}")
        return printedResult
    }

}